azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/MacUtils.java

```diff
@@ -0,0 +1,139 @@
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MacUtils {
+
+    private static final Pattern INTACT_MAC_PATTERN = Pattern.compile("^([0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2}$");
+    private static final Pattern MAC_PATTERN = Pattern.compile("([0-9A-Fa-f]{2}[:-]){5}[0-9A-Fa-f]{2}");
+    private static final String[] INVALID_MAC_ADDRESS = {"00:00:00:00:00:00", "ff:ff:ff:ff:ff:ff", "ac:de:48:00:11:22"};
+    private static final String[] UNIX_COMMAND = {"/sbin/ifconfig -a || /sbin/ip link"};
+    private static final String[] WINDOWS_COMMAND = {"getmac"};
+
+    public static String getMac() {
+        final String commandMac = getMacByCommand();
+        if (StringUtils.isNotBlank(commandMac)) {
+            return commandMac;
+        }
+        return getMacByNetworkInterface();
+    }
+
+    public static String getMacByCommand() {
+        List<String> macs = getMacsByCommand();
+        return CollectionUtils.isNotEmpty(macs) ? macs.get(0) : StringUtils.EMPTY;
+    }
+
+    private static List<String> getMacsByCommand() {
+        List<String> macs = new ArrayList<>();
+        final StringBuilder ret = new StringBuilder();
+        try {
+            final String os = System.getProperty("os.name").toLowerCase();
+            final String[] command = StringUtils.startsWithIgnoreCase(os, "win") ?
+                    WINDOWS_COMMAND : UNIX_COMMAND;
+            final ProcessBuilder probuilder = new ProcessBuilder(command);
+            final Process process = probuilder.start();
+            try (final InputStream inputStream = process.getInputStream();
+                 final InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
+                 final BufferedReader br = new BufferedReader(inputStreamReader)) {
+                String tmp;
+                while ((tmp = br.readLine()) != null) {
+                    ret.append(tmp);
+                }
+            }
+            if (process.waitFor() != 0) {
+                throw new IOException(String.format("Command %s execute fail.", String.join(" ", command)));
+            }
+        } catch (IOException | InterruptedException ex) {
+            return macs;
+        }
+        String commandMacsString = ret.toString();
+
+        Matcher matcher = MAC_PATTERN.matcher(commandMacsString);
+        while (matcher.find()) {
+            String mac = matcher.group(0);
+            if (isValidMac(mac)) {
+                macs.add(mac);
+            }
+        }
+        return macs;
+    }
+
+    public static String getMacByNetworkInterface() {
+        List<String> macs = getMacsByNetworkInterface();
+        if (CollectionUtils.isEmpty(macs)) {
+            return StringUtils.EMPTY;
+        }
+        return macs.get(0);
+    }
+
+    private static List<String> getMacsByNetworkInterface() {
+        List<String> macs = new ArrayList<>();
```

Comment:
same as above

Reply:
There are another (getMacByCommand as line 80) method to do that.

Reply:
as discussed in the meeting, made it as private.